apps/notifications/src/panel/utils/noticon2gridicon.js

```diff
@@ -1,27 +1,18 @@
-/**
- * External dependencies
- */
-import { get } from 'lodash';
-
-export const noticon2gridicon = ( c ) =>
-	get(
-		{
-			'\uf814': 'mention',
-			'\uf300': 'comment',
-			'\uf801': 'add',
-			'\uf455': 'info',
-			'\uf470': 'lock',
-			'\uf806': 'stats-alt',
-			'\uf805': 'reblog',
-			'\uf408': 'star',
-			'\uf804': 'trophy',
-			'\uf467': 'reply',
-			'\uf414': 'warning',
-			'\uf418': 'checkmark',
-			'\uf447': 'cart',
-		},
-		c,
-		'info'
-	);
+const noticon2gridicon = ( c ) =>
+	( {
```

Comment:
Should we declare that object only once at the top, and then reference it in the `noticon2gridicon` function?

Reply:
I debated as to which would be the better form here, but I don't have a strong opinion either way. I'm assuming that declaring the object once would be better for performance?

Reply:
Yes, it's better for performance, mostly theoretically - it's hard to measure its real impact in an application 😉 

To elaborate, the idea is to optimize for JS's memory management. JS's garbage collection algorithms would generally allocate memory when creating objects and free that memory once those objects are not used. With that in mind, let's take an example, which is again a bit dry and theoretical:

* We call `noticon2gridicon` 5 times in the same component. 
* That creates 5 copies of that object, each consuming X amount of memory. 
* JS's garbage collection may anytime between or after the 5 instantiations, meaning JS might need up to 5 times X the memory for storing the objects before garbage collection kicks in.
* If we had declared the object just once outside the function, that would have only needed X amount of memory, regardless of how many times we call `noticon2gridicon()`.

Again, that's hard to measure and mostly theoretical, but then again, declaring mostly static objects outside of functions vs inside of them is pretty low-cost in terms of developer work needed for it 😉